### ISSUE TITLE
Safari can't open seb(s) links, but second time it works

### ISSUE THREAD

**Issue by a-camacho** [SafeExamBrowser/seb-mac#70]
Dear developer team,

We are seeing a strange thing in recent versions of Mac OS X.
If you click on the button to start exam (seb(s) link) Safari gives you an error (attached).

![Capture d’écran 2020-09-29 à 17 52 27](https://user-images.githubusercontent.com/29976133/94582731-e10f2b00-027c-11eb-97fd-d2485d2c952c.png)

What is weird, is that if you reload the webpage (reload button or CMD+R) Safari suggests you to open that configuration file with Safe Exam Browser.

In other words, first click gives error, and then refresh starts the exam.

Do you have any information about that ?

Thank you very much.

**Comment by danschlet** [comment]
Unfortunately Apple has changed how Safari works with URLs which contain a non-standard protocol scheme, in this case seb(s). The Moodle Deeper Integration feature generates the SEB configuration file for a quiz dynamically, when the “Launch Safe Exam Browser” button is clicked. That button invokes the PHP script which generates the config file, its URL is "sebs://safeexambrowser.moodle.ch/lms/mod/quiz/accessrule/seb/config.php”, but when the script generates the config file, it redirects to the according .seb file. Safari 14 doesn’t like that, I guess because websites could use this mechanism to launch a potentially malicious application.

I currently don’t know if I could do anything on the side of the SEB application, to maybe make Safari “trust” the sebs:// protocol, I will research that. I guess it might also be possible to change the implementation of the “Lauch Safe Exam Browser” button in the Moodle Deeper Integration plugin, I can discuss that with its developers.

I guess at the moment the only workaround is to tell students to either use another browser than Safari or to manually invoke the reload, as you mentioned.

**Comment by danschlet** [comment]
This was fixed in recent versions of the Moodle SEB integration.
